web-interface/api/logic.py

```diff
@@ -270,35 +270,26 @@ def GetVersionInfo():
         revisionData = json.loads(url.read().decode())
         availableVersion = revisionData['CurrentVersion']
 
-    canUpdate = False
     availableVersionSplit = availableVersion.split('.')
     currentVersionSplit = currentVersion.split('.')    
-    if int(availableVersionSplit[0]) > int(currentVersionSplit[0]):
-        canUpdate = True
-    else:
-        if int(availableVersionSplit[0]) == int(currentVersionSplit[0]):
-            if int(availableVersionSplit[1]) > int(currentVersionSplit[1]):
-                canUpdate = True
 
     updateBranchName = 'master'
     updateBranchFile = '/media/usbdata/rpms/config/update-branch.txt'
     if os.path.isfile(updateBranchFile):
         file = open(updateBranchFile, 'r')
         updateBranchName = file.read()
 
-    # Always update if override-branch has been given, even if versions don't match   
-    developmentVersionOverride = False
-    if updateBranchName != 'master':
-        canUpdate = True
-        developmentVersionOverride = True
+    # Always update if override-branch has been given, even if versions don't match
+    isVersionOverridden = updateBranchName != 'master'
+    canUpdate = int(availableVersionSplit[0]) > int(currentVersionSplit[0]) or (int(availableVersionSplit[0]) == int(currentVersionSplit[0]) and int(availableVersionSplit[1]) > int(currentVersionSplit[1])) or isVersionOverridden
```

Comment:
Version-check is done already above. Is obslete in this place

Reply:
During refactoring of developmentVersionOverride to isVersionOverridden I had to see what changes the code and canUpdate was very odd assigned behind an if-statement. I was able to reduce code and improve readability of the code by assigning canUpdate once. So now the code is not only more readable, but also shorter and faster.